pywr/recorders/recorders.py

```diff
@@ -518,4 +519,16 @@ def finish(self):
         self._arrays = {}
         self._routes_flow_array = None
 
+    @staticmethod
+    def generate_dataframes(h5file, time='/time', scenarios='/scenarios'):
```

Comment:
Is there an argument that this should be a normal method? Would be quite nice to be able to use this on instances of the recorder directly

Reply:
It was written as a static method so you don't have to make an instance of the recorder, and therefore model, when post-processing the results. I've realised this function doesn't set the column index when reading the data which is kind of annoying when there are multiple scenarios.

Reply:
yeah, I didn't notice that the scenario func arg is not used. Looks like you will also get an error if you use it on a file with no time node. Maybe just use the default Pandas range index in these cases.

Reply:
To be honest it needs some tests!

Reply:
@Batch21 I've added a test and improved the function. Can you check?

Reply:
looks good